### ISSUE TITLE
Missing config. files?

### ISSUE THREAD

**Issue by KarlSemaan** [ryanlayer/giggle#68]
Hello,
I am facing the following error after having properly installed giggle on Windows through Ubuntu:

karl_semaan@DESKTOP-70LI237:~/giggle$ make
configure.ac: error: required file 'config.sub' not found
configure.ac: error: required file 'config.guess' not found
configure.ac:   try running autoreconf --install
cd lib/htslib; ./configure --disable-bz2 --disable-lzma --enable-libcurl
configure: error: cannot find required auxiliary files: config.guess config.sub
make: *** [Makefile:11: htslib] Error 1

It looks like there is a couple missing files in the directory that should normally be present in order to properly run the makefile.
Any idea how I can get this solve?

**Comment by ryanlayer** [comment]
Can you try going to lib/htslib and running autoreconf --install

**Comment by KarlSemaan** [comment]
Thank you for your reply!

**Comment by KarlSemaan** [comment]
Thank you for your reply!

**Comment by KarlSemaan** [comment]
Thank you for your reply! Just ran autoreconf -- install from lib/htslib, and then make from /giggle: I am getting another type of error I am sharing below:

karl_semaan@DESKTOP-70LI237:~/giggle$ cd lib/htslib
karl_semaan@DESKTOP-70LI237:~/giggle/lib/htslib$ autoreconf --install
karl_semaan@DESKTOP-70LI237:~/giggle/lib/htslib$ cd ../..
karl_semaan@DESKTOP-70LI237:~/giggle$ make
cd lib/htslib; ./configure --disable-bz2 --disable-lzma --enable-libcurl
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking for ranlib... ranlib
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking shared library type... plain .so
checking build system type... x86_64-pc-linux-gnu
checking host system type... Invalid configuration `unknown-Linux': machine `unknown' not recognized
configure: error: /bin/bash ./config.sub unknown-Linux failed
make: *** [Makefile:11: htslib] Error 1

Thanks again!

**Comment by ryanlayer** [comment]
There is something funny about the environment you are trying to compile
in. Can you describe it?

**Comment by KarlSemaan** [comment]
I am using Ubuntu terminal on Windows10. I am not sure about the environment, I did not specified it.
if I type "env" I get:
karl_semaan@DESKTOP-70LI237:~$ env
SHELL=/bin/bash
LIBGL_ALWAYS_INDIRECT=1
WSL2_GUI_APPS_ENABLED=1
WSL_DISTRO_NAME=Ubuntu
NAME=DESKTOP-70LI237
PWD=/home/karl_semaan
LOGNAME=karl_semaan
HOME=/home/karl_semaan
LANG=C.UTF-8
WSL_INTEROP=/run/WSL/191_interop
WAYLAND_DISPLAY=wayland-0
SSH_CONNECTION=127.0.0.1 60816 127.0.0.1 22007
TERM=xterm-256color
USER=karl_semaan
DISPLAY=localhost:10.0
SHLVL=1
XDG_RUNTIME_DIR=/mnt/wslg/runtime-dir
SSH_CLIENT=127.0.0.1 60816 22007
WSLENV=
XDG_DATA_DIRS=/usr/local/share:/usr/share:/var/lib/snapd/desktop
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/Program Files/WindowsApps/MicrosoftCorporationII.WindowsSubsystemForLinux_1.2.0.0_x64__8wekyb3d8bbwe:/mnt/c/Users/MDKARL~1/AppData/Roaming/MobaXterm/slash/bin:/mnt/c/Windows/:/mnt/c/Windows/system32/:/mnt/c/Program Files (x86)/Common Files/Oracle/Java/javapath:/mnt/c/Program Files/Addinsoft/XLSTAT/:/mnt/c/Program Files/Eclipse Adoptium/jdk-17.0.5.8-hotspot/bin:/mnt/c/Windows/system32:/mnt/c/Windows:/mnt/c/Windows/System32/Wbem:/mnt/c/Windows/System32/WindowsPowerShell/v1.0/:/mnt/c/Program Files/kallisto:/mnt/c/Program Files/kb_python:/mnt/c/Program Files/FastQC:/mnt/c/Program Files/PuTTY/:/mnt/c/Program Files/Go/bin:/mnt/c/Program Files/Docker/Docker/resources/bin:/mnt/c/Program Files/Git/cmd:/mnt/c/Users/MD Karl Semaan/AppData/Local/Programs/Python/Python311/Scripts/:/mnt/c/Users/MD Karl Semaan/AppData/Local/Programs/Python/Python311/:/mnt/c/Users/MD Karl Semaan/miniconda3:/mnt/c/Users/MD Karl Semaan/miniconda3/Library/mingw-w64/bin:/mnt/c/Users/MD Karl Semaan/miniconda3/Library/bin:/mnt/c/Users/MD Karl Semaan/miniconda3/Scripts:/mnt/c/Users/MD Karl Semaan/AppData/Local/Microsoft/WindowsApps:/mnt/c/Windows/sysnative/:/snap/bin
SSH_TTY=/dev/pts/3
HOSTTYPE=x86_64
PULSE_SERVER=unix:/mnt/wslg/PulseServer
OLDPWD=/home/karl_semaan/giggle
_=/usr/bin/env

Thank you for you help!

**Comment by wge11** [comment]
I encountered with the same problem. Then I tried this following:

> Can you try going to lib/htslib and running autoreconf --install

It reports another error:
`autoreconf: error: 'configure.ac' is required
`
Do you have any suggestions on solving this?
Thanks in advance!

**Comment by shokrof** [comment]
Hi,
I encountered the same issue here. I went to lib/htslib and executed  autoreconf --install. it finished without errors. I tried to run make from the main folder after and I got this error

```
checking host system type... Invalid configuration `unknown-Linux': machine `unknown' not recognized
configure: error: /bin/bash ./config.sub unknown-Linux failed
```

**Comment by mchowdh200** [comment]
It seems this issue has arisen in newer versions of Ubuntu (and perhaps other distributions).  Perhaps due to newer versions of build tools like autoconf, or related to configuration in the operating system itself.  I recently ran into this same issue on Ubuntu 22.04, but reverted to a machine with 18.04 and the issue vanished.  We'll have to investigate this further and pinpoint what the exact cause here is.

**Comment by mchowdh200** [comment]
For those of you who are still following this issue.  I've made a simple singularity def file that can be used to build and run giggle using a set of dependencies that work.

**giggle.def**
```
Bootstrap: docker
From: ubuntu:18.04

%post
    apt-get -y update
    apt-get install -y gcc make autoconf zlib1g-dev libbz2-dev libcurl4-openssl-dev \
                       libssl-dev ruby git
    git clone https://github.com/ryanlayer/giggle.git
    cd giggle
    make
    cp bin/giggle /bin

%runscript
    giggle "$@"
```
I used singularity community edition version 3.8.0.  To build the image from the def run
```
singularity build --fakeroot giggle.sif giggle.def
```
Then we can run giggle with the usual command line arguments by using `singularity run giggle.sif $args`.  Of course we need to bind some local directories to use for input/output with the singularity container's environment.  To do this we can use the `--bind` option:
```
singularity run --bind '/data:/data' giggle.sif index -i '/data/my_bed_files/*.gz' -o '/data/my_index'
```
In this example, I had a directory called `/data` where I had my bed files and where my desired index location was.  `--bind '/data:/data'` binds our local directory `/bind` to the container's own `/bind` directory.  To bind multiple locations just separate your `LOCAL_PATH:CONTAINER_PATH` pairs with commas.
